File: solutions/zig/darts/1/darts.zig
pub const Coordinate = struct {
    x: f32,
    y: f32,
    pub fn init(x: f32, y: f32) Coordinate {
        return Coordinate{ .x = x, .y = y };
    }
    pub fn score(self: Coordinate) usize {
        const lengthSquared = self.x * self.x + self.y * self.y;
        if (lengthSquared > 100.0) {
            return 0;
        }
        if (lengthSquared > 25.0) {
            return 1;
        }
        if (lengthSquared > 1.0) {
            return 5;
        }
        return 10;
    }
};

File: solutions/go/darts/1/darts.go
package darts

func Score(x, y float64) int {
	d := x*x+y*y
    if d <= 1.0 {
        return 10
    }
    if d <= 25.0 {
        return 5
    }
    if d <= 100.0 {
        return 1
    }
    return 0
}
